src/script/components/Modals/PrimaryModal/PrimaryModalTypes.ts

```diff
@@ -68,18 +74,20 @@ export enum PrimaryModalType {
 
 export interface ModalContent {
   checkboxLabel: string;
+  closeBtnTitle?: string;
   closeFn: () => void;
   closeOnConfirm?: boolean;
+  copyPassword?: boolean;
   currentType: string;
   inputPlaceholder: string;
   messageHtml: string;
-  messageText: string;
+  messageText: React.ReactNode;
   modalUie: string;
   onBgClick: () => void;
+  passwordGenerator?: boolean;
```

Comment:
This seems a little too specific. The Modal component should not be aware that there is a password generator that exists. I think this API can be made more generic

Reply:
this one is removed, agreed its unnecessary.